### PR TITLE
Adds an upsert vector function

### PR DESCRIPTION
**Pull request by alexthomas93**
# Description

This PR adds an `upsert_vector` utility function. This function wraps a Cypher query which allows users to attach a single vector to the property of a specific node. Unit tests have also been added for this function.

## Type of Change
- [X] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Medium

## How Has This Been Tested?
- [X] Unit tests
- [X] E2E tests
- [X] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [X] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [X] New files have copyright header
- [X] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: src/neo4j_genai/indexes.py [hunk 1]
```diff
@@ -143,3 +144,41 @@ def drop_index_if_exists(driver: neo4j.Driver, name: str) -> None:
...
+        node_id (int): The id of the node.
+        vector_prop (str): The name of the property to store the vector in.
+        vector (list[float]): The vector to store.
+
+    Raises:
+        Neo4jInsertError: If upserting of the vector fails.
+    """
+    try:
+        query = f"""
+        MATCH (n: {node_label})
```
Comment (oskarhane) [inline src/neo4j_genai/indexes.py:171]: No need for the label if you have the id.

Comment (alexthomas93) [reply src/neo4j_genai/indexes.py]: Good point! I've removed it

File: src/neo4j_genai/exceptions.py [hunk 2]
```diff
@@ -61,6 +61,12 @@ class Neo4jIndexError(Neo4jGenAiError):
     pass
 
 
+class Neo4jInsertionError(Neo4jGenAiError):
```
Comment (willtai) [inline src/neo4j_genai/exceptions.py:64]: I think we can also add this Exception to the docs

Comment (alexthomas93) [reply src/neo4j_genai/exceptions.py:64]: I've updated `docs/source/api.rst` with it. Is there anything else I need to do to update the docs?

Comment (willtai) [reply src/neo4j_genai/exceptions.py:64]: Nope this looks great!

**Comment by willtai** [comment]
@CodiumAI-Agent /update_changelog

**Comment by willtai** [comment]
This is looking good, please add an update the CHANGELOG and it's all good with me :)

**Review by oskarhane — APPROVED** [review]
Feel free to merge after changelog update 🧣

**Comment by alexthomas93** [comment]
@CodiumAI-Agent /update_changelog

**Comment by oskarhane** [comment]
@CodiumAI-Agent /update_changelog

**Pull request merged**